### PR TITLE
Added Gloo v0.0.20200317

### PR DESCRIPTION
**Pull request by stemann**
Referenced by Torch v1.6.0: https://github.com/pytorch/pytorch/tree/v1.6.0/third_party

### CODE REVIEW

**Comment by stemann** [comment]
Windows support should be fixed in #4571

File: G/Gloo/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+platforms = expand_cxxstring_abis(platforms)
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libgloo", :libgloo),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("LibUV_jll", v"2.0.0"),
```
Comment (giordano) [inline G/Gloo/build_tarballs.jl:43]: It doesn't seem to link to libuv:
```console
% readelf -d x86_64-linux-gnu-cxx11/lib/libgloo.so|head

Dynamic section at offset 0xd7e40 contains 26 entries:
  Tag        Type                         Name/Value
 0x0000000000000001 (NEEDED)             Shared library: [libpthread.so.0]
 0x0000000000000001 (NEEDED)             Shared library: [libstdc++.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libm.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libgcc_s.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [ld-linux-x86-64.so.2]
 0x000000000000000e (SONAME)             Library soname: [libgloo.so]
```
It's probably doing static linking?  I see lots of `uv_*` symbols in this library.
```suggestion
    BuildDependency("LibUV_jll", v"2.0.0"),
```

Comment (giordano) [reply G/Gloo/build_tarballs.jl]: Yeah, Julia's fork of libuv has inconsistent versioning

Comment (stemann) [reply G/Gloo/build_tarballs.jl]: Right, I remember reading something about the Julia being significant patches ahead/beside of main libuv.

**Comment by stemann** [comment]
> > Windows support should be fixed in #4571
> 
> Re-considering this - would be nice to have at least x86_64-w64 included...

Done :-)

**Pull request merged**